### DISCUSSION TITLE
mAP - Ignoring false positives?

### DISCUSSION THREAD

**Discussion by ChristofferEdlund** [Lightning-AI/torchmetrics#1753] [CompVision]
Hey.

When exploring torch metrics mAP we noticed that passing a GT with one box and a prediction with multiple boxes the mAP metric seem to ignore the false positives when calculating precision.

Is this the sought behavior of the mAP metric or am I employing it wrong?
(See image for example)

![image](https://user-images.githubusercontent.com/18189030/235704019-35cdab09-ae59-4830-877a-2d559ff3a0ff.png)

Code to reproduce:

```
from torchmetrics.detection.mean_ap import MeanAveragePrecision
import torch

map_metric = MeanAveragePrecision()

tm_gt = [{"boxes" :torch.tensor([[30, 30, 40, 40]]), "labels": torch.tensor([1]), "scores": torch.tensor([1])}]
tm_preds = [{"boxes": torch.tensor([[30, 30, 40, 40], [30, 25, 40, 35], [10, 10, 20, 20]]), "scores":torch.tensor([.8, .8, .8]), "labels":torch.tensor([1, 1, 2])}]


map_metric.forward(tm_preds, tm_gt)
```

output:

```
{'map': tensor(1.),
 'map_50': tensor(1.),
 'map_75': tensor(1.),
 'map_small': tensor(1.),
 'map_medium': tensor(-1.),
 'map_large': tensor(-1.),
 'mar_1': tensor(1.),
 'mar_10': tensor(1.),
 'mar_100': tensor(1.),
 'mar_small': tensor(1.),
 'mar_medium': tensor(-1.),
 'mar_large': tensor(-1.),
 'map_per_class': tensor(-1.),
 'mar_100_per_class': tensor(-1.)}
```
